src/GafferScene/Set.cpp

```diff
@@ -192,75 +194,105 @@ IECore::ConstInternedStringVectorDataPtr Set::computeSetNames( const Gaffer::Con
 {
 	ConstInternedStringVectorDataPtr inNamesData = inPlug()->setNamesPlug()->getValue();
 
-	InternedString name = namePlug()->getValue();
-	if( !name.string().size() )
+	const std::string &names = namePlug()->getValue();
+	if( !names.size() )
 	{
 		return inNamesData;
 	}
 
-	const std::vector<InternedString> &inNames = inNamesData->readable();
-	if( std::find( inNames.begin(), inNames.end(), name ) != inNames.end() )
+	if( modePlug()->getValue() == Remove )
 	{
 		return inNamesData;
 	}
 
-	if( modePlug()->getValue() == Remove )
+	InternedStringVectorDataPtr result = NULL;
+
+	Tokenizer namesTokenizer( names, boost::char_separator<char>( " " ) );
+	const std::vector<InternedString> &inNames = inNamesData->readable();
+	std::vector<InternedString>::const_iterator inBegin = inNames.begin();
+	std::vector<InternedString>::const_iterator inEnd = inNames.end();
+	for( Tokenizer::iterator name = namesTokenizer.begin(), end = namesTokenizer.end(); name != end; name++ )
```

Comment:
Since it seems we can end up with pretty large lists of sets, I'm a bit concerned about using `find()` (a linear search) inside this loop - the `numberOfNames * numberOfSets` number of comparisons could explode quite quickly. I wonder if we would be better to copy the input, tokenize directly onto the end of it and then use `sort`, `unique` and `erase` to get rid of duplicates?